arroyo-types/src/lib.rs

```diff
@@ -274,12 +274,56 @@ pub enum UpdatingData<T: Data> {
 }
 
 #[derive(Clone, Encode, Decode, Debug, Serialize, Deserialize, PartialEq)]
+#[serde(try_from = "DebeziumShadow<T>")]
 pub struct Debezium<T: Data> {
     before: Option<T>,
     after: Option<T>,
     op: DebeziumOp,
 }
 
+// Use a shadow type to perform post-deserialization validation that the expected fields
+// are set; see https://github.com/serde-rs/serde/issues/642#issuecomment-683276351
+#[derive(Clone, Encode, Decode, Debug, Serialize, Deserialize, PartialEq)]
+struct DebeziumShadow<T: Data> {
+    before: Option<T>,
+    after: Option<T>,
+    op: DebeziumOp,
+}
+
+impl<T: Data> TryFrom<DebeziumShadow<T>> for Debezium<T> {
+    type Error = &'static str;
+
+    fn try_from(value: DebeziumShadow<T>) -> Result<Self, Self::Error> {
+        match value.op {
+            DebeziumOp::Create => {
+                if value.after.is_none() {
+                    return Err("`after` must be set for Debezium create messages");
+                }
+            }
+            DebeziumOp::Update => {
+                if value.before.is_none() {
+                    return Err("`before` must be set for Debezium update messages; you may need to set REPLICA IDENTIFY FULL on your database");
```

Comment:
I think this is postgres specific behavior. Worth calling out?

Reply:
yeah, I'll add that -- unfortunately no way to know here what the database is